docs/aws/ContinuousIntegration/WebApp.rst

```diff
@@ -0,0 +1,110 @@
+
+Running end-to-end tests during development
+*******************************************
+
+The end-to-end tests run against an instance of the :ref:`nRF Asset Tracker for AWS <index_aws>`.
+
+Use either the credentials you created, :ref:`when setting up the solution <aws-getting-started>`, or enable the Web App CI feature and use the dedicated credentials created for this task.
+The latter option is the recommended approach since it limits the permissions to only the ones needed. 
+You can also use the credentials to :ref:`run the end-to-end tests on GitHub Actions <aws-continuous-integration-web-app-github>`.
+
+1. Add the following environment variables to your :file:`.envrc` file:
+
+   .. parsed-literal::
+       :class: highlight
+
+       # add to .envrc
+       export AWS_REGION="*your preferred AWS region*"
+       export AWS_ACCESS_KEY_ID="*value of Access key ID*"
+       export AWS_SECRET_ACCESS_KEY="*value of Secret access key*"
+       export WEBAPP_STACK_NAME="*name of your web application stack*"
+
+#. Run the following command to allow the changed :file:`.envrc` file:
+
+   .. code-block:: bash
+
+     direnv allow
+
+#. You can then run the tests using the following command:
+
+   .. code-block:: bash
+
+     npx playwright test
+
+Playwright Inspector
+--------------------
+
+For developing tests, it is helpful to run the `Playwright Inspector <https://playwright.dev/docs/inspector>`_.
+
+To enable the inspector during the tests, run:
+
+.. code-block:: bash
+
+  PWDEBUG=1 npx playwright test
+
+.. _aws-continuous-integration-web-app-github:
+
+Running end-to-end tests using GitHub Actions
+*********************************************
+
+`The Test and Release workflow <https://github.com/NordicSemiconductor/asset-tracker-cloud-app-aws-js/blob/41705dae8a5d4d7067c023297a3d38a2f0d1106e/.github/workflows/test-and-release.yaml>`_ runs the end-to-end tests for every commit.
+For this to work, you need a running instance of :ref:`nRF Asset Tracker for AWS <index_aws>`.
+The tests run against this instance.
+Typically, it is the production instance, to ensure that the web application works with the current production setup.
+Follow the :ref:`Getting started guide <aws-getting-started>` to set up the instance to be used for the tests.
+
+The test runner needs to interact with the instance to retrieve the app configuration and to provide test data.
+For this to be possible, you need to configure AWS credentials as `GitHub environment secrets <https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-an-environment>`_.
+
+1. Enable the web application CI of the nRF Asset Tracker for AWS:
+
+   .. code-block:: bash
+
+     node cli configure context stack web-app-ci 1
+
+#. Acquire the secrets using the nRF Asset Tracker for AWS CLI:
+
+   .. code-block:: bash
+
+     node cli web-app-ci -s
+
+#. Set the secrets:
+
+  a. Set the secrets using the GitHub UI:
```

Comment:
As these two are alternative, they should probably be just bullets, not ordered list.